src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -32,6 +48,221 @@
+        to None.
+
+        exit_on_error (bool, optional): Whether to exit execution on the first encountered error.
+        Defaults to True.
+
+        fallback_return_value (Union[Unset, Any], optional): The fallback return value for tasks
+        that encounter errors. Defaults to _unset.
+    """
+
+    def __init__(
+        self,
+        generation_fn: Callable[[Any], Any],
+        tqdm_bar_format: Optional[str] = None,
+        exit_on_error: bool = True,
+        fallback_return_value: Union[Unset, Any] = _unset,
+    ):
+        self.generate = generation_fn
+        self.fallback_return_value = fallback_return_value
+        self.tqdm_bar_format = tqdm_bar_format
+        self.exit_on_error = exit_on_error
+
+        self._TERMINATE = False
+
+    def _signal_handler(self, signum: int, frame: Any) -> None:
+        tqdm.write("Process was interrupted. The return value will be incomplete...")
+        self._TERMINATE = True
+
+    def run(self, inputs: Sequence[Any]) -> List[Any]:
+        signal.signal(signal.SIGINT, self._signal_handler)
+        outputs = [self.fallback_return_value] * len(inputs)
+        progress_bar = tqdm(total=len(inputs), bar_format=self.tqdm_bar_format)
+
+        for index, input in enumerate(inputs):
+            if self._TERMINATE:
+                break
+            try:
+                result = self.generate(input)
+                outputs[index] = result
+                progress_bar.update()
+            except Exception as e:
+                tqdm.write(f"Exception in worker: {e}")
+                if self.exit_on_error:
+                    break
+                else:
+                    progress_bar.update()
+        return outputs
+
+
+def get_executor_on_sync_context(
+    sync_fn: Callable[[Any], Any],
+    async_fn: Callable[[Any], Coroutine[Any, Any, Any]],
+    concurrency: int = 3,
+    tqdm_bar_format: Optional[str] = None,
+    exit_on_error: bool = True,
+    fallback_return_value: Union[Unset, Any] = _unset,
+) -> Union[AsyncExecutor, SyncExecutor]:
+    async_executor = AsyncExecutor(
+        async_fn,
+        concurrency=concurrency,
+        tqdm_bar_format=tqdm_bar_format,
+        exit_on_error=exit_on_error,
+        fallback_return_value=fallback_return_value,
+    )
+    sync_executor = SyncExecutor(
+        sync_fn,
+        tqdm_bar_format=tqdm_bar_format,
+        exit_on_error=exit_on_error,
+        fallback_return_value=fallback_return_value,
+    )
+    try:
+        asyncio.get_running_loop()
+        if getattr(asyncio, "_nest_patched", False):
+            return async_executor
+        else:
+            logger.warning(
+                "If running llm_classify inside a notebook, patching the event loop with "
+                "nest_asyncio will allow asynchronous eval submission, and is significantly "
+                "faster. To patch the event loop, run `nest_asyncio.apply()`."
+            )
```

Comment:
Don't we actually know it's running in a notebook here basically?

Also this text gets a bit swallowed. Might prefix with an emoji ℹ️ or 🤔

Reply:
I agree, the warning isn't super noticeable. Maybe 🐌 emoji.

Reply:
Easy to miss.

<img width="1013" alt="Screenshot 2023-11-21 at 10 28 12 AM" src="https://github.com/Arize-ai/phoenix/assets/15664869/ffd50540-39c8-43f2-95c7-00ff20c861f7">

Reply:
the snail emoji is great

Reply:
> Don't we actually know it's running in a notebook here basically?
> 
> Also this text gets a bit swallowed. Might prefix with an emoji ℹ️ or 🤔

@mikeldking it's most likely they're in a notebook given our audience but it isn't necessarily true